### ISSUE TITLE
how to use datadir link dir or making cross-seed search sqlite database

### ISSUE THREAD

**Issue by goffy59** [cross-seed/cross-seed#436]
On Qbittorrent I am using sqlite instead of the fastresume/etc so using BT_backup seems pointless at this point. Since I don't see a way to have it search my sqlite db I was hoping to setup data based searching like ive seen mentioned around here. This is normally working pretty good, I had deleted the old config and started fresh so I can setup download based searching instead of the old way. I would like to thank anyone who provides any helpful pointers to get this working. Thanks!

Tried to read the website but it was not very helpful. says 0 torrents searched here is my docker run:

docker run -d \
--name=cross-seed \
-e PGID=1000 -e PUID=1000 \
-e TZ=America/Phoenix \
-v /home/docker/cross-seed:/config \
-v /raid0/privatetorrent/downloads:/datadirs \
-v /raid0/privatetorrent/downloads:/linkdirs \
-v /home/docker/privatetorrent/config/qBittorrent/BT_backup:/torrents \
-p 2468:2468 \
--restart=always \
crossseed/cross-seed daemon

here is my config:

cat config.js 
"use strict";
// If you find yourself always using the same command-line flag, you can set
// it here as a default.
Object.defineProperty(exports, "__esModule", { value: true });
module.exports = {
    /**
     * Pause at least this much in between each search. Higher is safer.
     * It is not recommended to set this to less than 2 seconds.
     */
    delay: 10,
    /**
     * List of Torznab URLs.
     * For Jackett, click "Copy RSS feed"
     * For Prowlarr, click on the indexer name and copy the Torznab Url, then append "?apikey=YOUR_PROWLARR_API_KEY"
     * Wrap each URL in quotation marks, and separate them with commas.
     */
    torznab: ["http://atm:9696/136/api?apikey=redacted",
	"http://atm:9696/143/api?apikey=redacted",
	"http://atm:9696/121/api?apikey=redacted",
	"http://atm:9696/130/api?apikey=redacted",
	"http://atm:9696/114/api?apikey=redacted",
	"http://atm:9696/135/api?apikey=redacted",
	"http://atm:9696/117/api?apikey=redacted",
	"http://atm:9696/118/api?apikey=redacted",
	"http://atm:9696/132/api?apikey=redacted",
	"http://atm:9696/145/api?apikey=redacted",
	"http://atm:9696/144/api?apikey=redacted",
	"http://atm:9696/115/api?apikey=redacted",
	"http://atm:9696/120/api?apikey=redacted",
	"http://atm:9696/140/api?apikey=redacted",
	"http://atm:9696/141/api?apikey=redacted",
	"http://atm:9696/119/api?apikey=redacted",
	"http://atm:9696/147/api?apikey=redacted",
	"http://atm:9696/133/api?apikey=redacted",
	"http://atm:9696/142/api?apikey=redacted"
],
    /**
     * To search with downloaded data, you can pass in directories to your downloaded torrent data
     * to find matches rather using the torrent files themselves for matching.
     */
    dataDirs: "/datadirs",
    /**
     * Determines flexibility of naming during matching. "safe" will allow only perfect name matches
     * using the standard matching algorithm. "risky" uses filesize as its only comparison point.
     * Options: "safe", "risky"
     */
    matchMode: "safe",
    /**
     * Defines what category torrents injected by data-based matching should use.
     * Default is "cross-seed-data"
     */
    dataCategory: undefined,
    /**
     * If this is specified, cross-seed will create links to scanned files in the specified directory.
     * It will create a different link for every changed file name or directory structure.
     */
    linkDir: "/linkdirs",
    /**
     * cross-seed will use links of this type to inject data-based matches into your client.
     * Only relevant if dataDirs is specified.
     * Options: "symlink", "hardlink"
     */
    linkType: "symlink",
    /**
     * Whether to skip recheck in Qbittorrent. If using "risky" matchMode it is HIGHLY
     * recommended to set this to false.
     * Only applies to data based matches.
     */
    skipRecheck: false,
    /**
     * Determines how deep into the specified dataDirs to go to generate new searchees.
     * Setting this to higher values will result in more searchees and more API hits to
     * your indexers.
     */
    maxDataDepth: 10,
    /**
     * Directory containing torrent files.
     * For rtorrent, this is your session directory
     * as configured in your .rtorrent.rc file.
     * For deluge, this is ~/.config/deluge/state.
     * Don't change this for Docker.
     * Instead set the volume mapping on your docker container.
     */
    torrentDir: "/torrents",
    /**
     * Where to put the torrent files that cross-seed finds for you.
     * Don't change this for Docker.
     * Instead set the volume mapping on your docker container.
     */
    outputDir: "/cross-seeds",
    /**
     * Whether to search for single episode torrents
     */
    includeEpisodes: true,
    /**
     * Include torrents which contain non-video files
     * This option does not override includeEpisodes.
     * To search for everything except episodes, use (includeEpisodes: false, includeNonVideos: true)
     * To search for everything including episodes, use (includeEpisodes: true, includeNonVideos: true)
     */
    includeNonVideos: true,
    /**
     * fuzzy size match threshold
     * decimal value (0.02 = 2%)
     */
    fuzzySizeThreshold: 0.02,
    /**
     * Exclude torrents first seen more than this long ago.
     * Format: https://github.com/vercel/ms
     * Examples:
     * "10min"
     * "2w"
     * "3 days"
     */
    excludeOlder: undefined,
    /**
     * Exclude torrents which have been searched
     * more recently than this long ago.
     * Examples:
     * "10min"
     * "2w"
     * "3 days"
     */
    excludeRecentSearch: undefined,
    /**
     * With "inject" you need to set up one of the below clients.
     * Options: "save", "inject"
     */
    action: "save",
    /**
     * The url of your rtorrent XMLRPC interface.
     * Only relevant with action: "inject".
     * Could be something like "http://username:password@localhost:1234/RPC2
     */
    rtorrentRpcUrl: undefined,
    /**
     * The url of your qBittorrent webui.
     * Only relevant with action: "inject".
     * Supply your username and password inside the url like so:
     * "http://username:password@localhost:8080"
     */
    qbittorrentUrl: "http://user:pass@host:port",
    /**
     * The url of your Transmission RPC interface.
     * Usually ends with "/transmission/rpc".
     * Only relevant with action: "inject".
     * Supply your username and password inside the url like so:
     * "http://username:password@localhost:9091/transmission/rpc"
     */
    transmissionRpcUrl: undefined,
    /**
     * qBittorrent-specific
     * Whether to inject using categories with the same save paths as your normal categories.
     * Example: if you have a category called "Movies",
     * this will automatically inject cross-seeds to "Movies.cross-seed"
     */
    duplicateCategories: false,
    /**
     * cross-seed will send POST requests to this url
     * with a JSON payload of { title, body }.
     * Conforms to the caronc/apprise REST API.
     */
    notificationWebhookUrl: undefined,
    /**
     * Listen on a custom port.
     */
    port: 2468,
    /**
     * Bind to a specific host address.
     * Example: "127.0.0.1"
     * Default is "0.0.0.0"
     */
    host: undefined,
    /**
     * Run rss scans on a schedule. Format: https://github.com/vercel/ms
     * Set to undefined or null to disable. Minimum of 10 minutes.
     * Examples:
     * "10min"
     * "2w"
     * "3 days"
     */
    rssCadence: undefined,
    /**
     * Run searches on a schedule. Format: https://github.com/vercel/ms
     * Set to undefined or null to disable. Minimum of 1 day.
     * If you have RSS enabled, you won't need this to run often (2+ weeks recommended)
     * Examples:
     * "10min"
     * "2w"
     * "3 days"
     */
    searchCadence: undefined,
    /**
     * Fail snatch requests that haven't responded after this long.
     * Set to null for an infinite timeout.
     * Format: https://github.com/vercel/ms
     * Examples:
     * "30sec"
     * "10s"
     * "1min"
     * null
     */
    snatchTimeout: undefined,
    /**
     * Fail search requests that haven't responded after this long.
     * Set to null for an infinite timeout.
     * Format: https://github.com/vercel/ms
     * Examples:
     * "30sec"
     * "10s"
     * "1min"
     * null
     */
    searchTimeout: undefined,
};
//# sourceMappingURL=config.template.docker.cjs.map

**Comment by goffy59** [comment]
yeah but basically if i could get the download based searching working, it wouldnt matter about SQLITE? anyone have any examples of how to configure the download based data searching??

**Comment by goffy59** [comment]
did not mean to close this.

**Comment by goffy59** [comment]
please reopen this!!!!

**Comment by onedr0p** [comment]
@goffy59 From what was mention [here](https://github.com/cross-seed/cross-seed/issues/365#issuecomment-1420190362) it doesn't look possible yet? Also you can close this issue as it's a dupe of that issue.

**Comment by goffy59** [comment]
I am a little confused as to what you mean. I thought download based searching works?

**Comment by onedr0p** [comment]
Maybe this helps
https://gist.github.com/akesher123/aa91c71a1c37cdd4198bb9f63959ae53
